### PR TITLE
Introduce the NavigationBar page

### PR DESCRIPTION
**Pull request by bijington**
Provides the documentation for https://github.com/CommunityToolkit/Maui/pull/1678

### CODE REVIEW

**Review by pictos — COMMENTED** [review]
It looks great, just a small comment regarding the code sample

File: docs/maui/platform-specific/navigation-bar.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+```csharp
+using CommunityToolkit.Maui.PlatformConfiguration.AndroidSpecific;
+
+class NavigationBarPage : ContentPage
+{
+    public NavigationBarPage()
+    {
+        NavigationBar.SetColor(this, Colors.DarkSlateBlue);
+        NavigationBar.SetStyle(this, NavigationBarStyle.LightContent);
```
[the remark this answers is not in the document]
Comment (bijington) [reply docs/maui/platform-specific/navigation-bar.md:96]: Thanks I can update the usings. If we do this in C# should we use On inside the XAML as well?

[the remark this answers is not in the document]
Comment (pictos) [reply docs/maui/platform-specific/navigation-bar.md:96]: When we use that xmlns is the same idea as `On<Android>()` in C#

[the remark this answers is not in the document]
Comment (bijington) [reply docs/maui/platform-specific/navigation-bar.md]: Thanks for explaining

**Review by jfversluis — APPROVED** [review]

**Pull request merged**